docs/quickstart.md

```diff
@@ -3,196 +3,186 @@ title: Quickstart
-  --hiveconf hive.input.format=org.apache.hadoop.hive.ql.io.HiveInputFormat \
-  --hiveconf hive.stats.autogather=false \
-  --hiveconf hive.aux.jars.path=/path/to/packaging/hudi-hive-bundle/target/hudi-hive-bundle-0.4.6-SNAPSHOT.jar
+The [DataGenerator](https://github.com/apache/incubator-hudi/blob/master/hudi-spark/src/main/java/org/apache/hudi/QuickstartUtils.java) 
+can generate sample inserts and updates based on the the sample trip schema [here](#sample-schema)
 
-```
 
-### Run Hive Sync Tool
-Hive Sync Tool will update/create the necessary metadata(schema and partitions) in hive metastore. This allows for schema evolution and incremental addition of new partitions written to.
-It uses an incremental approach by storing the last commit time synced in the TBLPROPERTIES and only syncing the commits from the last sync commit time stored.
-Both [Spark Datasource](writing_data.html#datasource-writer) & [DeltaStreamer](writing_data.html#deltastreamer) have capability to do this, after each write.
+## Write data {#inserts}
+Generate sample records and load them into a DataFrame. Write the DataFrame into the Hudi table as below.
 
 ```
-cd hudi-hive
-./run_sync_tool.sh
-  --user hive
-  --pass hive
-  --database default
-  --jdbc-url "jdbc:hive2://localhost:10010/"
-  --base-path tmp/hoodie/sample-table/
-  --table hoodie_test
-  --partitioned-by field1,field2
+val inserts = convertToStringList(dataGen.generateInserts(10)) 
+val df = spark.read.json(spark.sparkContext.parallelize(inserts, 2));
+df.write.format("org.apache.hudi").options(getQuickstartWriteConfigs).option(PRECOMBINE_FIELD_OPT_KEY, "ts").option(RECORDKEY_FIELD_OPT_KEY, "uuid").option(PARTITIONPATH_FIELD_OPT_KEY, "partitionpath").option(TABLE_NAME, tableName).mode(Overwrite).save(basePath);
+``` 
+You can check the data generated under `/tmp/hudi_cow_table1/<region>/<country>/<city>/`
 
-```
-For some reason, if you want to do this by hand. Please 
-follow [this](https://cwiki.apache.org/confluence/display/HUDI/Registering+sample+dataset+to+Hive+via+beeline).
+**Modelling a record in Hudi:**
 
+Hudi depends on a record key (`uuid` in [schema](#sample-schema)), partition field (`region/county/city`) and combine logic (`ts` in [schema](#sample-schema)) to handle duplicates. For more info refer to [Modeling data stored in Hudi](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=113709185#Frequentlyaskedquestions(FAQ)-HowdoImodelthedatastoredinHudi?)
 
-### HiveQL {#hive}
 
-Let's first perform a query on the latest committed snapshot of the table
+**Ways to ingest data into Hudi:**
+- ***DeltaStreamer:*** For ingesting changes from external sources like Kafka, tailing DFS, or even other Hudi datasets.
+- ***Hudi Datasource:*** Capture data from a custom source using the Spark datasource API and write into Hudi
+For more info on using these, refer to [Writing Hudi Datasets](https://hudi.apache.org/writing_data.html)
 
+ 
+## Read data {#query}
```

Comment:
does it make sense to have these sections as just comments on the spark shell .. 

```
scala> // Now lets do some updates 
```

Reply:
I thought it would be like one big blob of spark shell commands if we chose comments over sections. Splitting them would make them readable. What do you think ?